File: Solution/WebSolution/AssessmentPortal/Microservices/EvaluationCriteriaAPI/Services/EvaluationCriteriaService.cs
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Services.Interfaces;
using Transflower.TFLAssessment.Repositories.Interfaces;

namespace Transflower.TFLAssessment.Services
{
    public class EvaluationCriteriaService : IEvaluationCriteriaService
    {
        private readonly IEvaluationCriteriaRepository _repository;
        
        public EvaluationCriteriaService(IEvaluationCriteriaRepository repository)
        {
            _repository = repository;
        }

        public async Task<bool> UpdateSubject(int id, int subjectId)
        {
            return await _repository.UpdateSubject(id, subjectId);
        }

        public async Task<bool> InsertCriteria(EvaluationCriteria criteria)
        {
            return await _repository.InsertCriteria(criteria);
        }

        public async Task<bool> UpdateCriteria(int evaluationCriteriaId, int questionId)
        {
            return await _repository.UpdateCriteria(evaluationCriteriaId, questionId);
        }
    }
}

File: Solution/Java/tflassessment-demo/src/main/java/com/transflower/tflassessment/demo/services/EvaluationCriteriaService.java
package com.transflower.tflassessment.demo.services;

import com.transflower.tflassessment.demo.entities.EvaluationCriteria;

public interface EvaluationCriteriaService {

    boolean updateSubject(int id, int subjectId);

    boolean insertCriteria(EvaluationCriteria criteria);

    boolean updateCriteria(int EvaluationCriteriaId, int subjectId);

}
